### PR TITLE
Add fail-only mode

### PR DESCRIPTION
**Pull request by davidot**
Also fix that progress bars appear in the middle of verbose results

The progress bar in the middle of verbose used to look like:
```
 17%|█████▉                             | 73/434 [00:01<00:05, 68.68tests/s]✅ .../libjs-test262/test262/test/built-ins/Array/prototype/slice/create-revoked-proxy.js

{}

 17%|█████▉                             | 74/434 [00:01<00:05, 68.68tests/s]✅ .../libjs-test262/test262/test/built-ins/Array/prototype/slice/create-species-undef.js

{}
```

But now looks like:
```
✅ /home/david/code/libjs-test262/test262/test/built-ins/Array/prototype/slice/create-revoked-proxy.js                                                                                                                                                                              

{}

✅ /home/david/code/libjs-test262/test262/test/built-ins/Array/prototype/slice/create-species-null.js                                                                                                                                                                               

{}
```
With the progressbar only all the way at the bottom

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -356,7 +359,10 @@ def run(self) -> None:
             for future in concurrent.futures.as_completed(futures):
                 test_run = future.result()
                 self.count_result(test_run)
-                if self.verbose:
+                if self.verbose or (self.fail_only and test_run.result != TestResult.PASSED):
```
Comment (linusg) [inline main.py:362]: Perhaps we should also exclude skipped tests here? I wouldn't consider them "failing".

**Review by linusg — COMMENTED** [review]
Please reformat the code with [black](https://github.com/psf/black) :)

File: main.py (hunk 1 above)
Comment (davidot) [reply main.py]: Good point I added `NON_FAIL_RESULTS = [TestResult.PASSED, TestResult.SKIPPED]` at the top to be able to specify the non failing results (for this mode only).

**Comment by davidot** [comment]
Reformatted the code and now also hide skipped tests

**Pull request merged**